ADA/ada.qmd

```diff
@@ -19,57 +17,194 @@ It brings together:
+
+---
+
+#### Writing and running code
+
+---
+
+In the Databricks interface, you have the choice to use notebooks, scripts or the Databricks SQL Editor. 
+
+Notebooks are made up of multiple code or markdown cells. They are different to IDEs (Integrated Development Environments) like RStudio. You can read more about the differences between IDEs and notebooks and where each might be useful [on the NHS-R website](/ADA/ada.html#differences-between-ides-and-notebooks).
+
+Scripts (e.g. R scripts) can be created and saved in your Workspace and run via the Databricks interface. 
+
+There is detailed guidance on using the Databricks SQL editor, which is similar to SSMS, on the [ADA guidance site](https://analytical-data-access.education.gov.uk/get-support/using-sql-in-databricks-sql-editor).
+
+One benefit of Databricks is the use of workflows to run your code. Workflows allow you to automate the running of single or multiple scripts, so if you have large amounts of data processing to do on a regular basis, you can set up a workflow to complete this overnight. Your laptop does not need to be switched on for this to work. 
+
+You can also run code from RStudio or other IDEs locally and execute it against the data held in Databricks. To do this, you will need to set up a connection between RStudio and the Databricks SQL Warehouse. This is explained in the [R scripts in RStudio section](/ADA/ada.html#r-scripts-in-rstudio) below.
+
+---
+
+#### Accessing code from a git repository
+
+Databricks has inbuilt git functionality, and you can save your notebooks and scripts inside git repositories as with any other file formats. There is guidance on using Databricks with Azure DevOps on our [Databricks and version control page](ADA/git_databricks.html). 
+
+---
+
+## What this means for existing code
+
+### SQL scripts
+
+---
+
+#### Scripts currently used in SSMS
+
+If you currently use data and scripts stored in SSMS for your statistics publication pipeline, there will be small changes to your production pipelines that vary depending on your current process. 
+
+Your SQL code will need to be migrated over to Databricks to allow it to read from the Databricks SQL Warehouse and will require some minor changes to syntax as SSMS uses T-SQL and Databricks uses Spark SQL. The ADA team are currently developing a tool to allow you to easily convert T-SQL queries to Spark SQL syntax. 
+
+You can write new SQL code in the Databricks SQL Editor, which is very similar to SSMS. 
+
+---
+
+#### SQL scripts in repositories or embedded in R scripts
+
+These scripts do not need to be moved, but the syntax of the scripts or embedded code will need to be rewritten from T-SQL to Spark SQL. The only thing you will need to change is the connection source to allow the code to run. Rather than pointing to data held in current SQL databases, you will need to point towards the Databricks Delta Lake instead. 
```

Comment:
Probably a stupid question but... Is this specific to if you're running these via databricks? 
Or is the need to write using Spark necessary for any connection to their database?

Reply:
Necessary for any connection I believe, you need to use Spark SQL to talk to the delta lake (data storage) regardless of where you do it from?

Reply:
I had a conversation with @cjrace about this last week because my brain melted when I tried to think it through so I don't think it is a stupid question! Cam is right though

Reply:
Think this bit still needs a slight tweak:

> These scripts do not need to be moved, but the syntax of the scripts or embedded code will need to be rewritten from T-SQL to Spark SQL. The only thing you will need to change is the connection source to allow the code to run. Rather than pointing to data held in current SQL databases, you will need to point towards the Databricks Delta Lake instead.

would make it this instead

> These scripts do not need to be moved, but the syntax of the scripts or embedded code will need to be rewritten from T-SQL to Spark SQL.
> You will also need to change is the connection source to allow the code to run. Rather than pointing to data held in current SQL databases, you will need to point towards the Databricks Delta Lake instead.